File: 3751-total-waviness-of-numbers-in-range-i/solution.cpp
#include <string>

class Solution {
private:
    // Helper function to calculate waviness of a single number
    int getWaviness(int n) {
        std::string s = std::to_string(n);
        int len = s.length();
        if (len < 3) {
            return 0;
        }
        
        int waviness = 0;
        for (int i = 1; i < len - 1; ++i) {
            if ((s[i] > s[i - 1] && s[i] > s[i + 1]) || 
                (s[i] < s[i - 1] && s[i] < s[i + 1])) {
                waviness++;
            }
        }
        return waviness;
    }

public:
    int sumWaviness(int num1, int num2) {
        int totalWaviness = 0;
        for (int i = num1; i <= num2; ++i) {
            totalWaviness += getWaviness(i);
        }
        return totalWaviness;
    }
};

File: 3751-total-waviness-of-numbers-in-range-i/solution.py
class Solution:
    def sumWaviness(self, num1: int, num2: int) -> int:
        total_waviness = 0
        
        for num in range(num1, num2 + 1):
            s = str(num)
            length = len(s)
            if length < 3:
                continue
            
            # Count peaks and valleys
            for i in range(1, length - 1):
                if (s[i] > s[i - 1] and s[i] > s[i + 1]) or (s[i] < s[i - 1] and s[i] < s[i + 1]):
                    total_waviness += 1
                    
        return total_waviness